lib/components/src/typography/DocumentFormatting.tsx

```diff
@@ -71,9 +72,15 @@ export const Pre = styled.pre<{}>(withReset, withMargin, ({ theme }) => ({
 }));
 
 const Link: FunctionComponent<any> = ({ href: input, children, ...props }) => {
+  // If storybook is hosted at a non-root path (e.g. `/storybook/`),
+  // the base url needs to be prefixed to storybook paths.
+  let storybookBaseUrl =
+    typeof window !== 'undefined' ? window.parent.document.location.pathname : '/';
```

Comment:
i believe `window.parent` causes a CORS error in composed storybooks cc @ndelangen

Reply:
The preview cannot 'hook' directly into the manager and vice versa.

Reply:
Thank you for the comments! Could you elaborate a bit what exactly the problem is and how we can solve it? I'm not really familiar with Storybook internals.

Reply:
`window.parent` is an illegal operation when the frame is on another origin @amannn

Reply:
Good point, thanks!

I just tried it, the precise error is:

> Uncaught DOMException: Blocked a frame with origin "http://localhost:8081" from accessing a cross-origin frame.

What would be a good strategy to handle this? Should we use try/catch to try to use the parent location and otherwise fall back to the default linking, which is based on the inner iframe?

I'm not sure how common it is that the inner iframe is on a separate domain. At least for the Storybook installation I'm maintaining it is the same one.

Reply:
@amannn You might want to give this a read:
https://storybook.js.org/docs/react/workflows/storybook-composition

gist: Storybooks composing other storybook's will result in the manager & iframe running on a different origin.

Reply:
A try-catch could solve this, but I'd like to better understand in what scenario you need the parent's parent path, why isn't the preview running on the same path for you?

Reply:
I've got different versions of a component library hosted at a path like this:

```
https://domain.tld/component-library/v72.2.0/?path=/docs/components-button--default
```

The content from the docs iframe is loaded at:

```
src="iframe.html?id=button--default&viewMode=docs"
```

Therefore when a link within the docs is clicked, you end up at …

```
https://domain.tld/component-library/v72.2.0/iframe.html?path=/docs/components-button--default
```

… whereas the correct path should be:

```
https://domain.tld/component-library/v72.2.0/?path=/docs/components-button--default
```

Reply:
@amannn could you setup a reproduction of this? perhaps?

Give me the ability to debug this locally?

Reply:
Yep, there is a way of testing this locally:

1. Build the `official-storybook`
2. Run `npx serve` in the project root
3. `open http://localhost:5000/built-storybooks/official-storybook/?path=/docs/addons-a11y-namedexportbutton--default`
4. Now when you click on the link to "BaseButton" in the docs, you end up at http://localhost:5000/built-storybooks/official-storybook/?path=/docs/addons-a11y-basebutton--default, which is the correct URL.

If you revert the changes I made to `lib/components/src/typography/DocumentFormatting.tsx`, rebuild Storybook core and the `official-storybook`, you'll see that now the link points to `http://localhost:5000/built-storybooks/official-storybook/iframe?path=/docs/addons-a11y-basebutton--default` (note the additional `iframe` part). This URL doesn't work.

Reply:
@ndelangen Did you have a chance to test this as described above?

Reply:
Is there anything I can do to help with this?

Reply:
I haven't had time to dive into this.. I understand the code changes in your PR fixes the issue, but it creates new ones, where CORS error occur. We need another way of fixing the issue. accessing window.parent, is not an option, unfortunately.

Reply:
Thanks for your reply! I've just rebased the branch and wrapped the logic with a try/catch as discussed above. As you mention, it doesn't fix the situation for composed storybooks though – only for non-composed ones.

Would it make sense to have this as a first fix and work on a more involved situation for composed storybooks separately? I currently don't know enough about the architecture of Storybook to propose a more general fix …

Reply:
@ndelangen I've rebased this PR based on the current `next` branch. Is there anything I could do to help get this merged? It seems like some unrelated CI checks are failing.